### DISCUSSION TITLE
Indonesia Market Daikin Compatibility

### DISCUSSION THREAD

**Discussion by NathanAdhitya** [revk/xESP32-Faikout#303] [General]
Hi, I wish to simply share my initial experiment results. Perhaps this would be useful for the other folks willing to try out this project or try it on their home Aircon.

I'm currently testing it on an FTKQ35SVM4. Upon opening the panel, the board has a S403 connector present.
The pinout was similar to other posts that mentioned the S403 connector. I was successfully able to connect Faikin to the AC. Faikin was able to obtain the home, liquid, and outside temps and was able to turn the AC on/off. The connector had a 2.54mm pitch. Regular Dupont connectors didn't fit so I had to modify a JST XH2.54 connector.

For this experiment, I used whatever material I had on hand. I used an ESP32 Dev board as the base, connected a 2N2222 with some resistors to handle 3.3V <-> 5V TX/RX, and a cheap step-down regulator. The RX side simply used a 1K and 2K voltage divider, the TX used a 2N2222 with a 10K pullup to the 5V.

I'm interested to know if there are other fellow people from the SEA region trying out this project. The models that are sold in the EU/UK seem to have different models from the ones sold here.

While my initial experiments were successful, I'll delay properly installing it as my janky work doesn't neatly fit inside the AC. Probably going to try to DIY a circuit board with affordable components that are easy to obtain here. If perhaps anyone has a more budget-friendly circuit board design for this project, I'm interested. The ESP32-S3-Mini seems a bit too expensive here. ESP-WROOM-32 seems much more obtainable.

Edit: Tried to open an FTC25NV14. S403 connector present. Dupont does not have enough friction. JST XH2.54 that fit in the FTKQ did not fit here.

**Comment by NathanAdhitya** [comment]
FTC25NV14 (non-inverter) has a S403 connector present. Connectors were custom crimped dupont connectors without housing. Heat shrink was applied between the bare connectors for insulation.

@QuartzAl helped out making a poor man's version of Faikin. We flashed the S1 firmware in commit ff7f59e92f83c2f27592c857016de2ab610826fd for an ESP-WROOM-32D and used GPIO pins 15 and 13 for TX and RX respectively.

We got it working (so far). We will post the schematics and PCB soon for those who might be interested.
We came out with a per-PCB cost of around ~60k IDR per module (~4 USD). PCB was assembled manually with components locally sourced. PCB were manufactured by JLCPCB.

**Reply by Tamsy** [reply]
Having 3 FTC25NV14 here I am very interested to learn about the schematics and the PCB for that model type. Can you please elaborate on how you did accomplish this?

**Reply by NathanAdhitya** [reply]
> Having 3 FTC25NV14 here I am very interested to learn about the schematics and the PCB for that model type. Can you please elaborate on how you did accomplish this?

Hi, you can refer to the PCB design and BOM here: https://oshwlab.com/stevmcg05/esp32-faikin. The current design works, but has one issue where there is a 16kHz hum from the buck converter. All components can be sourced from online shops e.g. from the iSee store in Tokopedia. You can also print the PCB at a local fab, or at JLCPCB, costing around 75k IDR after taxes for 5 boards.

**Reply by Tamsy** [reply]
Thank you for the quick reply, Nathan.
Is it the iSee Shop in Surabaya (through Toped) where I can get all the needed components?

Unfortunately I am a newbe when it comes to assembling electronics like PCB boards. Honestly speaking, https://oshwlab.com/stevmcg05/esp32-faikin tells me nothing. Maybe you know a shop or somebody here in Indonesia who can do the assembling or even sell the ready to use PCB to me? Taking 5 boards at once wouldn't be a problem. I'm in BSD (Serpong) btw.

**Reply by NathanAdhitya** [reply]
> Thank you for the quick reply, Nathan. Is it the iSee Shop in Surabaya (through Toped) where I can get all the needed components?
> 
> Unfortunately I am a newbe when it comes to assembling electronics like PCB boards. Honestly speaking, https://oshwlab.com/stevmcg05/esp32-faikin tells me nothing. Maybe you know a shop or somebody here in Indonesia who can do the assembling or even sell the ready to use PCB to me? Taking 5 boards at once wouldn't be a problem. I'm in BSD (Serpong) btw.

Hi, I'm currently in Surabaya. Unfortunately I do not know any shop that sell a ready-to-use Faikin yet. I have one fully assembled board that you can tinker with if you are interested. I don't remember all the sources of our components, but we mostly got them from iSee. We have no plans to sell it as a "product" yet before we iron out the issues with the current revision. If you are interested, send me your WhatsApp number through my public email, [REDACTED].

**Reply by Tamsy** [reply]
Hi Nathan. I just replied to you through email :)

**Comment by justinwyq** [comment]
Hi, I'm in Singapore and it's not easy to obtain a Faikin board here either. I am interested in creating one with an ESP32 dev kit. Could you send me the schematic and components you used? Thank you.

**Reply by revk** [reply]
They are all in this repository

**Reply by NathanAdhitya** [reply]
Hi, if you're starting off with a DevKit, the cheapest method would be:
1. Determine if your system outputs 5V or it outputs more than that. If it's beyond 5V, get a step down module. Find common ones that suit your needs and tune it to output 5V.
2. Find a suitable transistor for TX (remember, ESP32 I/O is usually not 5V tolerant, or well it might be, but for who knows how long). Use it as a logic converter.
3. Use a resistor divider network to step down the AC TX into MCU RX to 3.3V.
4. Connect everything together
5. Flash faikin
6. Profit???

You can build all of that on a perfboard with headers. I can't guarantee the perfboard will fit inside of your AC, but I can tell you that's how I tested mine.
Well, that's how I got started at least. That's the absolute bare minimum that you can get away with. Be aware that the Faikin that revk made has proper protections and is more resistant to stupidity. The method above may be cheap, but you have a higher chance of frying something.

If you can afford a PCB fab to produce some PCBs for you, we have a prototype that does what we described, but with a bare ESP32-WROOM-32/32D. Go to: [my previous comment](https://github.com/revk/ESP32-Faikin/discussions/303#discussioncomment-12596877) for more information.

Good luck.

**Reply by justinwyq** [reply]
Hi @NathanAdhitya, thanks to your helpful comments, I've managed to get it working with some locally available parts - ESP32C3, a logic converter and buck converter. I appreciate the support! Thank you @revk for maintaining this project.

**Comment by NathanAdhitya** [comment]
Hi @revk,
I have noticed that you have closed the discussion. Is it possible to keep it open instead?
I'd like to keep publishing my testing results here while keeping it visible for others.
Thanks.

**Reply by revk** [reply]
You can still post anyway, but if working it seemed closed.
